common/stream-model.cpp

```diff
@@ -779,177 +773,209 @@ namespace rs2
-                { RS2_FRAME_METADATA_AUTO_EXPOSURE                        , "Auto Exposure Mode indicator. Zero corresponds to AE switched off. " },
-                { RS2_FRAME_METADATA_WHITE_BALANCE                        , "White Balance setting as a color temperature. Kelvin degrees" },
-                { RS2_FRAME_METADATA_TIME_OF_ARRIVAL                      , "Time of arrival in system clock " },
-                { RS2_FRAME_METADATA_TEMPERATURE                          , "Temperature of the device, measured at the time of the frame capture. Celsius degrees " },
-                { RS2_FRAME_METADATA_BACKEND_TIMESTAMP                    , "Timestamp get from uvc driver. usec" },
-                { RS2_FRAME_METADATA_ACTUAL_FPS                           , "Actual hardware FPS. May differ from requested due to Auto-Exposure" },
-                { RS2_FRAME_METADATA_FRAME_LASER_POWER_MODE               , "Laser power mode. Zero corresponds to Laser power switched off and one for switched on." },
-                { RS2_FRAME_METADATA_EXPOSURE_PRIORITY                    , "Exposure priority. When enabled Auto-exposure algorithm is allowed to reduce requested FPS to sufficiently increase exposure time (an get enough light)" },
-                { RS2_FRAME_METADATA_POWER_LINE_FREQUENCY                 , "Power Line Frequency for anti-flickering Off/50Hz/60Hz/Auto. " },
-            };
+        float max_text_width = 0.;
+        for( auto && kvp : stream_details )
+            max_text_width = std::max( max_text_width, ImGui::CalcTextSize( kvp.name.c_str() ).x );
 
-            for (auto i = 0; i < RS2_FRAME_METADATA_COUNT; i++)
-            {
-                auto&& kvp = frame_md.md_attributes[i];
-                if (kvp.first)
-                {
-                    auto val = (rs2_frame_metadata_value)i;
-                    std::string name = rs2_frame_metadata_to_string(val);
-                    std::string desc = "";
-                    if (descriptions.find(val) != descriptions.end()) desc = descriptions[val];
-                    stream_details.push_back({ name, rsutils::string::from() << kvp.second, desc });
-                }
-            }
+        ImGui::SetCursorScreenPos( { stream_rect.x + 10, stream_rect.y } );
 
-            float max_text_width = 0.;
-            for (auto&& kvp : stream_details)
-                max_text_width = std::max(max_text_width, ImGui::CalcTextSize(kvp.name.c_str()).x);
+        std::string profile_id = rsutils::string::from() << "##Metadata-" << profile.unique_id();
+        ImGui::BeginChild( profile_id.c_str(), ImVec2( stream_rect.w - 8, stream_rect.h ) );
+        
+        // Draw metadata background
+        ImGui::GetWindowDrawList()->AddRectFilled(
+            { stream_rect.x - 5, stream_rect.y },
+            { stream_rect.x + max_text_width * 2,
+              stream_rect.y + ( stream_details.size() * ( ImGui::GetTextLineHeight() + 3.5f ) ) },
```

Comment:
1. Why do we double this `stream_rect.x + max_text_width * 2`
2. don't we need the start point  + offset to values column + max_text_width?
3. Why ` + 3.5f`?

Reply:
1. It was a simple way to get a size for the background. Today I rewrite the way to be dynamically calculated from values but it does not look good. (The background looks to be shaking)

Reply:
I refactor a variable to be constant but it is not affected the shaking bug, because the live scop of the variable is short.
Every time when we entering to the stream-model the variable changes and this causes a shaking.

Reply:
Waiting for the new code..
On your latest pic I still saw height differences between values and types..